pkg/config/config.go

```diff
@@ -118,6 +118,56 @@ func NewMicroshiftConfig() *MicroshiftConfig {
 	}
 }
 
+// Determine if the config file specified a NodeName (by default it's assigned the hostname)
+func (c *MicroshiftConfig) IsDefaultNodeName() bool {
+	hostname, err := os.Hostname()
+	if err != nil {
+		klog.Fatalf("Failed to get hostname %v", err)
+	}
+	return c.NodeName == hostname
+}
+
+// Read or set the NodeName that will be used for this MicroShift instance
+func (c *MicroshiftConfig) establishNodeName() (string, error) {
+	filePath := filepath.Join(GetDataDir(), "nodename")
```

Comment:
Hide this file, maybe? Users should not edit it.
It would be possible to stop the service, edit the file, change the config (in order to not use the default), start the service and have a broken node name, right?

Reply:
It's written with read-only mode for all (root/users)

May be we can add a first line as comment explaining that change is not supported, and that changing that will lead to problems.

Hiding can lead to the same issues, as soon as they find where to change it.

Reply:
You are right, missed the read only. Maybe add this to docs instead of the file? You would need to parse it.

Reply:
Right, I made it hidden, and I will explain in the docs that NodeName cannot be changed once established.